### PR TITLE
skip and restore resources with local-config annotation

### PR DESCRIPTION
**Pull request by mengqiy**
This PR currently only changes `set-labels`.
We can do the same thing for set-annotation, set-namespace and ensure-name-substring. `ProcessLocalConfigResources` can be reused.

In the above mentioned functions, we use [`NewResMapFromRNodeSlice`](https://pkg.go.dev/sigs.k8s.io/kustomize/api/resmap#Factory.NewResMapFromRNodeSlice) from kustomize repo to convert `[]*yaml.RNode` to `ResMap`.
NewResMapFromRNodeSlice drops resources with `local-config` annotation and there is no way to control this behavior: https://github.com/kubernetes-sigs/kustomize/blob/api/v0.8.9/api/resource/factory.go#L205-L236

We want to re-use kustomize's logic as much as possible since it's well-tested and mature. We do the preprocessing and post-processing in `ProcessLocalConfigResources`.

### CODE REVIEW

File: functions/go/set-labels/kustomizerhelper.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func ProcessLocalConfigResources(resources []*kyaml.RNode, fltr kio.Filter) ([]*kyaml.RNode, error) {
```
Comment (phanimarupaka) [inline functions/go/set-labels/kustomizerhelper.go:9]: Please add a godoc to explain what this method is doing.

Comment (phanimarupaka) [reply functions/go/set-labels/kustomizerhelper.go:9]: Why are we doing this in a function as opposed to orchestrator ? This is a common problem for all the functions correct ?

Comment (mengqiy) [reply functions/go/set-labels/kustomizerhelper.go:9]: This is PR is trying to fix a bug in functions that use `ResMap` which include set-annotations, set-labels, set-namespace and ensure-name-substring.
This bug affects `render`, for example https://github.com/GoogleContainerTools/kpt/tree/87191859fdef443395741d85e134e5aa1cd0f877/package-examples/pipeline-validate.

Input: 4 resources (2 normal resources + 2 local-config resources for gatekeeper)
==(set-label fn)==> 2 resources (2 normal resources)   # we expect the 2 local-config resources to be preserved
==(enforce-gatekeeper)==> 2 resources (2 normal resources) 

We can't really fix this bug in the orchestrator.

**Comment by frankfarzan** [comment]
I'm not following this change. Why do functions care about `local-config` annotation? Can you file an issue first.

**Comment by mengqiy** [comment]
Created https://github.com/GoogleContainerTools/kpt/issues/1933.
Any alternative fix is to make some change in upstream kustomize as mentioned in 1933

**Comment by mengqiy** [comment]
Closing this PR since we got https://github.com/kubernetes-sigs/kustomize/pull/3940

**Pull request closed without merging**